runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -98,6 +113,21 @@ local LanguageTree = {}
 
 LanguageTree.__index = LanguageTree
 
+---@param injection_query vim.treesitter.Query
+---@return boolean
+local function query_has_combined(injection_query)
```

Comment:
We may want to move this logic into `vim.treesitter.Query` and have it also calculate some other metadata when loading a query.

Reply:
what about doing this in `Query.new` and storing the result in a field? `query.has_combined`

Reply:
That's basically what I'm suggesting. Doesn't need to be done this PR but keep it in mind.

Reply:
done 
<https://github.com/neovim/neovim/compare/7b7736ced849114dcab5f3140e1565452d1ddb3d..1dc295cdb71d171b9f7bbf89a802e42db63225c6#diff-c9baf58a11444ecc08b1a8facc69b0bf9a8033d1b3a2771c6ccefd161e987b79>